### DISCUSSION TITLE
Lowercase FK when table uses CamelCase

### DISCUSSION THREAD

**Discussion by dekiesel** [pgmodeler/pgmodeler#1842] [Q&A] [answered]
Consider two tables, `TableA` and `TableB`.

When I create a relationship between those two pgmodeler creates a constraint `TableA_fk` on `TableB` and a column called `id_TableB`.

Is there a way to tell pgmodeler to create those objects in lowercase, i.e. `tablea_fk` and `id_tableb`?

I'd like my tablenames to be in CamelCase, but my columns to be in lowercase. Or is this bad practice?

**Comment by rkhaotix** [comment, marked as answer]
Hi @dekiesel 

I don't recommend using CamelCase for table names (or any other object in pgModeler). That's because the code generator will always double-quote the identifiers. For example, the table `TableA` will have the following code `CREATE TABLE public."TableA" ...`

That can be a real problem for you if you need to constantly type SQL commands to manage your database since PostgreSQL treats differently `TableA` and `"TableA"`. I recommend the following reading (especially the item 4.1.1):

https://www.postgresql.org/docs/current/sql-syntax-lexical.html

Anyway, to answer the question, you can adjust the names for any object generated by relationships but pgModeler does not change the string case automatically (maybe a future enhancement can add that). For more details see here (section 3.10.13):

https://www.pgmodeler.io/support/docs/310-relationships?v=1.1.0

**Reply by dekiesel** [reply]
I see, that is indeed a problem. Thank you very much for your help!
